### PR TITLE
Omarkj make standalone

### PR DESCRIPTION
**Pull request by omarkj**

### CODE REVIEW

File: src/folsom_cowboy.app.src [hunk 1]
```diff
@@ -4,38 +4,6 @@
...
-                  {"/_ping", folsom_cowboy_ping_handler, []},
-                  {"/_port", folsom_cowboy_port_handler, []},
-                  {"/_process", folsom_cowboy_process_handler, []},
-                  {"/_statistics", folsom_cowboy_statistics_handler, []},
-                  {"/_system", folsom_cowboy_system_handler, []},
-                  {"/_ets", folsom_cowboy_ets_handler, []},
-                  {"/_dets", folsom_cowboy_dets_handler, []}
-                 ]}]}
-        ]}
+  {applications, []},
```
Comment (ferd) [inline src/folsom_cowboy.app.src]: should always include `stdlib` and `kernel`. Should also add `jsx`.

File: src/folsom_cowboy_handler.erl [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+get_metrics(<<"true">>) ->
+    [{M, proplists:delete(tags, List)} ||
+        {M, List} <- folsom_metrics:get_metrics_info()];
+get_metrics(_) ->
+    folsom_metrics:get_metrics().
+
+metric_exists(Id) when is_list(Id) ->
+    metric_exists(list_to_binary(Id));
+metric_exists(Id) when is_binary(Id) ->
+    case folsom_metrics:metric_exists(Id) of
```
Comment (ferd) [inline src/folsom_cowboy_handler.erl:72]: Doesn't this put a folsom dependency on this app?

**Comment by ferd** [comment]
Damn that makes things a lot simpler. I have a few questions regarding dependencies, but otherwise +1.

File: rebar.config [hunk 3]
```diff
@@ -1,7 +1,18 @@
-{deps, [
-      {folsom, ".*", {git, "git://github.com/boundary/folsom", master}},
-      {cowboy, ".*", {git, "git://github.com/ninenines/cowboy", "1.0.x"}},
-      {ibrowse, ".*", {git, "git://github.com/cmullaparthi/ibrowse.git", {tag, "v2.2.0"}}}
-  ]}.
 {erl_opts, [debug_info]}.
 {cover_enabled, true}.
+
+
+{deps, [{jsx, [], {git,"git@github.com:talentdeficit/jsx.git",
```
Comment (evanmcc) [inline rebar.config:5]: folsom and cowboy should likely go in the real deps

**Comment by evanmcc** [comment]
+1, this is awesome.

**Comment by evanmcc** [comment]
also rebar3 shell works great for testing:

```
$ rebar3 shell
... junk
===> Compiling folsom_cowboy
Erlang/OTP 17 [erts-6.4] [source] [64-bit] [smp:8:8] [async-threads:0] [hipe] [kernel-poll:false] [dtrace]

Eshell V6.4  (abort with ^G)
1> application:ensure_all_started(folsom_cowboy).
```

and you're off to the races

**Comment by ferd** [comment]
@evanmcc https://github.com/rebar/rebar3/pull/473 will add stuff to auto-boot apps (if you have a relx tuple or use `{shell_apps, [folsom_cowboy]}`) and also supports calling `r3:do(compile)` which re-runs rebar3 compile and autoloads modified modules :)

**Pull request merged**